lib/policyEvaluator/evaluator.js

```diff
@@ -0,0 +1,257 @@
+    for (let i = 0; i < statementResource.length; i ++) {
+        // Handle variables (must handle BEFORE wildcards)
+        const policyResource =
+            substituteVariables(statementResource[i], requestContext);
+        // Handle wildcards
+        const arnSegmentsMatch =
+            checkArnMatch(policyResource, requestRelativeId,
+                requestResourceArr, true);
+        if (arnSegmentsMatch) {
+            return true;
+        }
+        continue;
+    }
+    // If no match found, no resource is applicable
+    return false;
+}
+
+/**
+ * Check whether action in policy statement applies to request
+ * @param {Object} requestAction - Type of client request
+ * @param {string | [string]} statementAction - Action(s) impacted
+ * by policy statement
+ * @return {boolean} true if applicable, false if not
+ */
+function isActionApplicable(requestAction, statementAction) {
+    if (!Array.isArray(statementAction)) {
+        // eslint-disable-next-line no-param-reassign
+        statementAction = [statementAction];
+    }
+    for (let i = 0; i < statementAction.length; i ++) {
+        // No variables in actions so no need to handle
+        const regExStrOfStatementAction =
+            handleWildcards(statementAction[i]);
+        const actualRegEx = new RegExp(regExStrOfStatementAction);
+        if (actualRegEx.test(requestAction)) {
+            return true;
+        }
+    }
+    // If no match found, return false
+    return false;
+}
+
+/**
+ * Check whether request meets policy conditions
+ * @param {Object} requestContext - info about request
+ * @param {Object} statementCondition - Condition statement from policy
+ * @return {boolean} true if meet conditions, false if not
+ */
+function meetConditions(requestContext, statementCondition) {
+    // The Condition portion of a policy is an object with different
+    // operators as keys
+    for (let i = 0; i < Object.keys(statementCondition).length; i ++) {
+        const operator = Object.keys(statementCondition)[i];
+        const hasIfExistsCondition = operator.endsWith('IfExists');
+        // If has "IfExists" added to operator name, find operator name
+        // without "IfExists"
+        const bareOperator = hasIfExistsCondition ? operator.slice(0, -8) :
+            operator;
+        const operatorsWithVariables = ['StringEquals', 'StringNotEquals',
+            'StringEqualsIgnoreCase', 'StringNotEqualsIgnoreCase',
+            'StringLike', 'StringNotLike', 'ArnEquals', 'ArnNotEquals',
+            'ArnLike', 'ArnNotLike'];
+        const operatorCanHaveVariables =
+            operatorsWithVariables.indexOf(bareOperator) > -1;
+        // Loop through conditions with the same operator
+        // Note: this should be the actual operator name, not the bareOperator
+        const conditionsWithSameOperator = statementCondition[operator];
+        for (let j = 0; j < Object.keys(conditionsWithSameOperator).length;
```

Comment:
It's not the length of the Object but the call itself is very expensive. You should cache that in a variable.